docs/tutorial.tex

```diff
@@ -140,12 +140,27 @@ \subsection{Custom Start-Up}
 
 \begin{verbatim}
 $7000 value buf
-: go buf s" myprogramfile" buf
+: go restore-forth
+buf s" myprogramfile" buf
 loadb buf - evaluate ;
 ' go start !
 save-forth @0:durexforth
 \end{verbatim}
 
+Each start routine should start by calling \texttt{restore-forth} to unpack the dictionary. Before \texttt{save-forth} writes the program to disk, it moves the header data down next to the code data. Without calling \texttt{restore-forth}, there will be no space for new words, and only 32 bytes between \texttt{here} and \texttt{latest}.
+
+\subsection{Turn-key Operation}
+
+If you have developed a program that has no further need of the interpreter, you can eliminate the dictionary entirely before calling \texttt{save-forth} with \texttt{ top to latest }. Now, the header space is 1 byte at \texttt{top}.
+
+As noted above, without calling \texttt{restore-forth}, \texttt{top} will remain 32 bytes above \texttt{here}, or else it will move to its original \texttt{top}. If you wish to keep the header space safe, reserve a byte that remains at value 0. For example, after you have assigned you \texttt{start} word:
+
+\begin{verbatim}
+here 0 , top to latest top! save-forth @0:turnkey
+\end{verbatim}
+
+This allows your program to use memory down to \texttt{here}.

```

Comment:
You know, all of this stuff seems maybe just too complicated.

It should really be enough with two words, `SAVE-FORTH FILE` versus `SAVE-PRG FILE STARTWORD` (or maybe `START-EXE` or whatever?). The system should be smart enough to take care of all the rest.

I understand the idea of compacting wordlist and word definitions for a smaller file. In practice though, is it really worth the extra complication? These days, most people anyway use fastloaders.

Reply:
The dictionary shrink came as a necessary feature for building the cart.  I like the idea of separating it into distinct words, the cart (and, by proxy, the release disk) created with `SAVE-PRG`.  I'll put this together.

Reply:
Oh - what I meant with `SAVE-PRG` is that it would be the strip-wordlist "turnkey" version.
For the cartridge, and the release disk, we would still need the wordlist.

Reply:
I have implemented `SAVE-PRG` without any wordlist stripping and left instructions about it in the manual.  I can factor out the part that's relevant for the initial build and automate wordlist stripping into `SAVE-PRG`, it's probably a good idea.

Reply:
I added SAVE-PACK as the third option, in case anyone doesn't have a fastloader ;)

I can hide it if you don't think it's useful in the dictionary.  Speaking of, should I include `HIDE` in base?